### PR TITLE
Feat(DPLAN-11858): splitStatement: adjust design for sidebar

### PR DESCRIPTION
**Pull request by sakutademos**
### Ticket
[DPLAN-11858](https://demoseurope.youtrack.cloud/issue/DPLAN-11858/FE-3-Als-Nutzerin-mochte-ich-eine-nutzerfreundliche-Benutzeroberflache-beim-Aufteilen-von-Stellungnahmen-vorfinden-auch-bei)

**Description:** The sidebar has to be adjusted to the new design. ([Prototype](https://www.figma.com/proto/3cu1AZiWm74SRinXCgLdSC/DPLAN-2085-Sidebar-SplitStatements?node-id=120-2476&t=953CokImURbxtBsC-1&scaling=min-zoom&content-scaling=fixed&page-id=120%3A2375&starting-point-node-id=120%3A3750) and [design template](https://www.figma.com/design/3cu1AZiWm74SRinXCgLdSC/DPLAN-2085-Sidebar-SplitStatements?node-id=120-2375&t=XKBdSkYGoo4lfdDD-0))

- calculate the available height between the header and footer, and apply a max-height to the sidebar to ensure it fits within the available space.

- add two floating context buttons with `position: absolute`. The top button is positioned at the bottom of the _Selected Tags_ section: since the `tags` section has `overflow-y: hidden` the button is not visible if it is placed into this section, even through `overflow-x` is visible.

- Handle varying widths of selected tags based on different cases (refer to the [design template](https://www.figma.com/design/3cu1AZiWm74SRinXCgLdSC/DPLAN-2085-Sidebar-SplitStatements?node-id=120-2375&t=XKBdSkYGoo4lfdDD-0)). Implement the `assignTagSizeClasses` function to check the position of each tag relative to the next tag element and apply appropriate size classes accordingly.

### Linked PRs 
- fix issue with the align prop in the DpButtonRow component --> [demosplan-ui](https://github.com/demos-europe/demosplan-ui/pull/911) 
- https://github.com/demos-europe/demosplan-core/pull/3456 - adjustments in core

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [x] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: client/js/components/statement/splitStatement/SideBar.vue [hunk 1]
```diff
@@ -8,107 +8,201 @@
...
+        v-if="editingSegment && editingSegment.tags.length > 0"
+        class="flex flex-wrap gap-1 bg-white">
+        <div
+          v-for="(tag, idx) in editingSegment.tags"
+          :key="`tag_${idx}`"
+          :class="assignTagSizeClasses(tag,idx)">
+          <div
+            :class="[
+              'tag flex whitespace-nowrap overflow-hidden text-sm px-0.5 py-0.5',
+              isTagAppliedToSegment(tag.id) ? 'bg-gray-500': 'bg-green-400',
```
Comment (spiess-demos) [inline client/js/components/statement/splitStatement/SideBar.vue:31]: In fact, the green background color was not meant to be static, it should fade into the default bg color for those tags. However for this to work with css alone, we will have to teach Tailwind something like this...

```
  theme: {
    extend: {
      keyframes: {
        'background-transition': {
          '0%': { backgroundColor: '#48bb78' }, // Tailwind green-500
          '100%': { backgroundColor: '#a0aec0' } // Tailwind gray-500
        }
      },
      animation: {
        'background-transition': 'background-transition 1s ease-in-out forwards'
      }
    }
  }
```
This, however, would be best done in demosplan-ui. I'm gonna look into it tomorrow.

Another question: right now you're using the default tailwind colors - i realize that is already possible as we're not yet configured tailwind with other colors. However i'd suggest not to do so, as those colors won't exist. If https://github.com/demos-europe/demosplan-ui/pull/932  is merged and released, `bg-status-complete` vs. `bg-status-neutral` could be used here.

File: client/js/components/statement/splitStatement/SideBar.vue [hunk 2]
```diff
@@ -8,107 +8,201 @@
...
-        :selected="selectedTags"
-        :place-holder="Translator.trans('tag.search')"
-        :options="searchableTags" />
-
-      <!-- create tags + topics -->
-      <dp-create-tag
-        v-else
-        @close-create-form="showCreateForm = false" />
+      <!--   Floating Context Button -->
+      <button
```
Comment (spiess-demos) [inline client/js/components/statement/splitStatement/SideBar.vue:52]: I'd propose to make this a local component, too. Or, would there be so many different props?

File: client/js/components/statement/splitStatement/SideBar.vue (hunk 1 above)
Comment (sakutademos) [reply client/js/components/statement/splitStatement/SideBar.vue:32]: what we discussed yesterday, here are some changes: https://github.com/demos-europe/demosplan-core/pull/3433/commits/27a4a72bb25868854ea3902179a019ed80358ed6

File: client/js/components/statement/splitStatement/SideBar.vue [hunk 3]
```diff
@@ -281,22 +431,38 @@ export default {
...
+      const header = document.getElementById('header')
+      const footer = document.getElementById('footer')
+
+      const viewportHeight = window.innerHeight
+      const headerHeight = header.offsetHeight
+      const footerHeight = footer.offsetHeight
+
+      const mainHeight = viewportHeight - headerHeight - footerHeight
+
+      return mainHeight * 0.8 // 80% from the main height
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/statement/splitStatement/SideBar.vue]: i have removed this part

File: client/js/components/statement/splitStatement/SideBar.vue [hunk 4]
```diff
@@ -281,22 +431,38 @@ export default {
 
       this.setProperty({ prop: 'editingSegment', val: segment })
       this.locallyUpdateSegments([this.editingSegment])
+    },
+
+    getSideBarMaxHeight () {
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/statement/splitStatement/SideBar.vue]: change it here, but the offset does not work as expected for me. i will look into it a bit later today: https://github.com/demos-europe/demosplan-core/pull/3433/commits/27a4a72bb25868854ea3902179a019ed80358ed6

File: client/js/components/statement/splitStatement/FloatingContextButton.vue [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    class="bg-white rounded shadow absolute p-0.5"
+    :aria-controls="section"
+    :aria-expanded="isContentCollapsed"
+    :data-cy="`sidebar:floatingContextButton:${section}`"
+    @click="toggleContentVisibility"
+    @mouseover="show"
+    @mouseleave="hide">
+    <dp-icon
+      :aria-label="Translator.trans('content.show/hide')"
+      class="w-4 h-4 rounded-sm text-blue-700 hover:bg-gray-100"
```
Comment (salisdemos) [inline client/js/components/statement/splitStatement/FloatingContextButton.vue:13]: @spiess-demos is this one of the colors, wich will not work anymore, after https://github.com/demos-europe/demosplan-ui/pull/932 finds its way into core? Then we should use something else from the beginning on.

Comment (spiess-demos) [reply client/js/components/statement/splitStatement/FloatingContextButton.vue:13]: yes, both `text-blue-700` and `bg-gray-100` are tailwind defaults. If we'd manage to release a new demosplan-ui version, and bump demosplan-core with it, we could use the correct colors here.

Comment (spiess-demos) [inline client/js/components/statement/splitStatement/FloatingContextButton.vue:13]:
```suggestion
      class="w-4 h-4 rounded-sm text-interactive hover:text-interactive-hover active:text-interactive-active hover:bg-interactive-subtle-hover active:bg-interactive-subtle-active"
```

**Review by spiess-demos — CHANGES_REQUESTED** [review]

**Review by spiess-demos — APPROVED** [review]
lgtm!

**Pull request merged**